compiler/codegen/library.cpp

```diff
@@ -243,37 +244,143 @@ static void setupPythonTypeMap() {
 
 }
 
+static void makePXDFile(std::vector<FnSymbol*> functions);
+static void makePYXFile(std::vector<FnSymbol*> functions);
+
 void codegen_library_python(std::vector<FnSymbol*> functions) {
   if (fLibraryCompile && fLibraryPython) {
     setupPythonTypeMap();
 
-    fileinfo pxd = { NULL, NULL, NULL };
+    strcpy(pxdName, "chpl_");
+    strncat(pxdName, libmodeHeadername,
+            sizeof(libmodeHeadername)-strlen(pxdName)-1);
```

Comment:
So the code you're writing has a lot of strcpy / strncat.
I tolerate some of these but in general I'd rather look at code that uses `+` on C++ strings (and then calls `.c_str()`) or uses our own `astr` method to concatenate strings.

Reply:
I've been using mostly to deal with the things that get set via compiler flags (mostly because I was following precedent).  I don't expect this to be set via a compiler flag, though, so I will definitely use a C++ string instead!